code/game/objects/items/devices/magnetic_lock.dm

```diff
@@ -80,220 +123,283 @@
 				emagcard.uses--
 				visible_message("<span class='danger'>[src] sparks and falls off the door!</span>", "<span class='danger'>You emag [src], frying its circuitry[status == STATUS_ACTIVE ? " and making it drop onto the floor" : ""]!</span>")
 
-				setstatus(STATUS_BROKEN)
+				status = STATUS_BROKEN
+				if (target)
+					detach()
+					update_icon()
 				return
 
-			if (status == STATUS_ACTIVE && istype(I, /obj/item/weapon/card/id))
-				if (check_access(I) && !constructionstate)
-					user << "<span class='notice'>You swipe your [I] through [src], making it drop onto the floor with a thud.</span>"
-					setstatus(STATUS_INACTIVE)
-					return
-				else if (constructionstate)
-					user << "<span class='danger'>You cannot swipe your [I] through [src] with it partially dismantled!</span>"
-					return
-				else
-					user << "<span class='danger'>A red light flashes on [src] as you swipe your [I] through it.</span>"
-					flick("deny",src)
+			if (iswelder(I))
+				var/obj/item/weapon/weldingtool/WT = I
+				if (WT.remove_fuel(2, user))
+					user.visible_message(span("notice", "[user] starts welding the metal shell of [src]."), span("notice", "You start [hacked ? "repairing" : "welding open"] the metal covering of [src]."))
+					playsound(loc, 'sound/items/Welder.ogg', 50, 1)
+					overlays += "overlay_welding"
+					if (do_after(user, 25, 1))
+						user << span("notice", "You are able to [hacked ? "repair" : "weld through"] the metal shell of [src].")
+						if (hacked) locked = 1
+						else locked = 0
+						hacked = !hacked
+						overlays -= "overlay_welding"
```

Comment:
All items in the list "overlays" are of class `image`, I do believe. Attempting to remove text from it like this surely doesn't work?

`overlay_welding` should be a `var/image/overlay_welding` for this to work.